File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        if (prices.Length == 0)
            return 0;
        int maxProfit = 0;
        int minPrice = prices[0];
        for (int i = 1; i < prices.Length; i++)
        {
            minPrice = Math.Min(minPrice, prices[i]);
            maxProfit = Math.Max(prices[i] - minPrice, maxProfit);
        }
        return maxProfit;
    }
}

File: Data Structures & Algorithms/insert-new-interval/submission-2.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        if not len(intervals):
            return [newInterval]
        res = []
        is_add_new = False
        for i in range(len(intervals)):
            interval = intervals[i]
            if interval[1] < newInterval[0]:
                res.append(interval)
                is_add_new = True
            elif interval[0] > newInterval[1]:
                res.append(newInterval)
                res.extend(intervals[i:])
                is_add_new = False
                break
            else:
                is_add_new = True
                newInterval[1] = max(interval[1], newInterval[1])
                newInterval[0] = min(interval[0], newInterval[0])
        if is_add_new:
            res.append(newInterval)
        return res